### ISSUE TITLE
Truncated data raises an IndexError, rather than a ValueError

### ISSUE THREAD

**Issue by alex** [wbond/asn1crypto#29]
I didn't see this documented, so I infer that it's probably a bug:

```pycon
>>>> from asn1crypto.algos import DSASignature
>>>> DSASignature.load("\x30")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/alex_gaynor/.virtualenvs/cryptography/site-packages/asn1crypto/core.py", line 229, in load
    value, _ = _parse_build(encoded_data, spec=spec, spec_params=kwargs, strict=strict)
  File "/Users/alex_gaynor/.virtualenvs/cryptography/site-packages/asn1crypto/core.py", line 4851, in _parse_build
    info, new_pointer = _parse(encoded_data, encoded_len, pointer)
  File "/Users/alex_gaynor/.virtualenvs/cryptography/site-packages/asn1crypto/parser.py", line 151, in _parse
    length_octet = ord(encoded_data[pointer]) if _PY2 else encoded_data[pointer]
IndexError: string index out of range
>>>> DSASignature.load(b"\x30\x03\x02\x00\x02", strict=True).native
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/alex_gaynor/.virtualenvs/cryptography/site-packages/asn1crypto/core.py", line 3434, in native
    self._parse_children(recurse=True)
  File "/Users/alex_gaynor/.virtualenvs/cryptography/site-packages/asn1crypto/core.py", line 3281, in _parse_children
    parts, child_pointer = _parse(self._contents, contents_length, pointer=child_pointer)
  File "/Users/alex_gaynor/.virtualenvs/cryptography/site-packages/asn1crypto/parser.py", line 151, in _parse
    length_octet = ord(encoded_data[pointer]) if _PY2 else encoded_data[pointer]
IndexError: string index out of range
```

**Comment by wbond** [comment]
Thanks for the bug report!

This should be fixed by 72e76c16fb4460a26d0f152c6579238b83654d31

**Comment by alex** [comment]
Awesome, thanks!
